### PR TITLE
Adding aseries for modified bessel functions

### LINKED ISSUE #26210: Asymptotic series expansions for modified bessel functions
The TODO from bessel.py says 
```
# TODO
# o Scorer functions G1 and G2
# o Asymptotic expansions
#   These are possible, e.g. for fixed order, but since the bessel type
#   functions are oscillatory they are not actually tractable at
#   infinity, so this is not particularly useful right now.
```

So most bessel type functions like Hankel, Kelvin or even Spherical Bessel functions would fall under this category but modified bessel functions don't.
**Besseli** grows exponentially whereas **Besselk** exponentially decays so we can have aseries implementations for these and also that would be in use for calculating limits through gruntz.

### PR DESCRIPTION
**Pull request by arnabnandikgp**
#### References to other Issues or PRs
Fixes #26210

#### Release Notes

* series
  * `_eval_aseries` method  was added for modified bessel functions.

### CODE REVIEW

File: sympy/functions/special/bessel.py [hunk 1]
```diff
@@ -586,6 +586,19 @@ def _eval_nseries(self, x, n, logx, cdir=0):
...
+        nu, z = self.args
+        point = args0[1]
+
+        if point is S.Infinity:
+            s = [(RisingFactorial(S.Half-nu,k)*RisingFactorial(
+                S.Half+nu,k))/((2)**(k)*z**((2*k+1)/2)*factorial(k)) for k in range(n)] + [Order(1/z**((2*n+1)/2), x)]
+            return (exp(z)/sqrt((2*pi))) * Add(*s)
+
+        return super()._eval_aseries(n, args0, x, logx)
+
```
Comment (anutosh491) [inline sympy/functions/special/bessel.py:601]: I think I had worked on a solution for this in my gsoc proposal a couple years back (though I wasn't able to implement it)
Would you like to have a look ? 
I think that might give you some direction for this PR. You can find my proposal [here](https://drive.google.com/file/d/1P8PFVoSyTcskkRVYmbHd-85o96dOiEnZ/view?pli=1)

Comment (arnabnandikgp) [reply sympy/functions/special/bessel.py:601]: If we fix the above decrepency with something like this 
```diff
--- a/sympy/core/expr.py
+++ b/sympy/core/expr.py
@@ -3504,7 +3504,10 @@ def as_coeff_exponent(self, x) -> tuple[Expr, Expr]:
         c, p = s.as_coeff_mul(x)
         if len(p) == 1:
             b, e = p[0].as_base_exp()
-            if b == x:
+            # if b == x:
+            if b in [x, 1/x]:
+                if b == 1/x:
+                    e = -e
                 return c, e
         return s, S.Zero

```
we have the test passing alright.

**Comment by arnabnandikgp** [comment]
the test that is failing is of the following simplified form 
```
In [3]: limit(besseli(1,sqrt(x)),x,oo)
Out[3]: besseli(1, 1/sqrt(_w))
```
It turns out the reason is in the following lines:
https://github.com/sympy/sympy/blob/0d6b9f64e12c2a01ea2720b0ca0415c05cbc77c5/sympy/series/gruntz.py#L554-L574
gruntz tries to find the leading term of `besseli(1, 1/sqrt(_w))` but is not able to do so even after utilising  the aseries expansion for the same,like it tries out `besseli(1, 1/sqrt(x)).series(x,0)` in line 563 and gets things like 
```
In [7]: sqrt(2)*x**0.25*exp(1/sqrt(x))/(2*sqrt(pi)) - 3*sqrt(2)*x**0.75*exp(1/sqrt(x))/(16*sqrt(pi))
Out[7]: 
          1                1 
          ──               ──
    0.25  √x         0.75  √x
√2⋅x    ⋅ℯ     3⋅√2⋅x    ⋅ℯ  
──────────── - ──────────────
    2⋅√π           16⋅√π     

```
the above do not yield a leadterm and thus at end we are returned `besseli(1, 1/sqrt(_w))` from `mrv_leadterm` which is alas returned as the final result. Such error oddly do not rise for limit(besseli(1,(x)**k),x,oo) where k is an integer as at such cases only mothods like `_eval_as_leading_term` suffices for the the result computation while for non-integer powers of x we have `aseries` called in and `exp(1/(x)**k)` and happens only when k is not an integer, maybe this has direct reason for the above decrepency of how using a right result from `as_coeff_exponent` may help in avoiding this error. Though I am still not sure how things were computed correctly before aseries was implemented i.e in the current master.

**Comment by arnabnandikgp** [comment]
Going thorough implementation of some error functions that have similar aseries implementation, I stumbled upon the following :
https://github.com/sympy/sympy/blob/07da3cec213b829d1edef980bd44b3547ffb0f8c/sympy/functions/special/error_functions.py#L2740-L2748
It seems that for gruntz to work correctly we require to pass tractable expression into it and thus may need to add a separate helper function for the modified bessel functions alike.

**Comment by anutosh491** [comment]
Hi @arnabnandikgp I see this PR is marked as draft. If you're done with the changes here, please mark it ready for review. Also get the CI to pass beforehand

**Comment by arnabnandikgp** [comment]
ping @anutosh491 for reviews

File: sympy/functions/special/tests/test_bessel.py [hunk 2]
```diff
@@ -142,6 +142,10 @@ def test_besseli_series():
         x**(S(7)/2)/144 + x**(S(9)/2)/2880 + x**(S(11)/2)/86400 + O(x**6)
     assert besseli(-2, sin(x)).series(x, n=4) == besseli(2, sin(x)).series(x, n=4)
 
+    #test for aseries
+    assert besseli(0,x).series(x, oo, n=4) == sqrt(2)*(sqrt(1/x) - (1/x)**(S(3)/2)/8 - \
+        3*(1/x)**(S(5)/2)/128 - 15*(1/x)**(S(7)/2)/1024 + O((1/x)**(S(9)/2), (x, oo)))*exp(x)/(2*sqrt(pi))
```
Comment (anutosh491) [inline sympy/functions/special/tests/test_bessel.py:147]: Do we have working results for `-oo` and cases like `I*oo` ?

Comment (arnabnandikgp) [reply sympy/functions/special/tests/test_bessel.py:147]: Yes, according to abramowitz and stegun and wolfram alpha we have the above expansion true for negative infinity too, I have added that in the latest commit. Though for I*oo cases I could not find any specific references anywhere.

**Comment by oscarbenjamin** [comment]
This looks okay to me.

**Comment by arnabnandikgp** [comment]
can you all also look into #26951, a very similar fix.

**Review by anutosh491 — APPROVED** [review]
Looks good to me.

**Pull request merged**